tex/alg-geom/sheaves.tex

```diff
@@ -37,8 +37,24 @@ \section{Pre-sheaves}
 Recall that $\OO_V$ took \emph{open sets of $V$} to \emph{rings},
 with the interpretation that $\OO_V(U)$ was a ``ring of functions''.
 
+Recall from \Cref{sec:sheaf_regular_functions} that $\OO_V$, as a set, consist
+of simply the algebraic functions. However, if we view $\OO_V$ purely as a set,
+the structure of the functions is essentially thrown way.
+
+Let us see how the functions in $\OO_V$ are related to each other:
+\begin{itemize}
+	\ii Each function in $\OO_V$ is defined on a open set $U \subseteq V$.
+	\ii If two functions are defined on the same open set, you can add and
+	multiply them together. In other words, $\OO_V(U)$ is a ring.
+	\ii Given a function $f \in \OO_V(U)$, we can restrict it to a smaller open
+	subset $W \subseteq U$.
+\end{itemize}
+These are the operations that we will impose on a pre-sheaf.
+
 \subsection{Usual definition}
 So here is the official definition of a pre-sheaf.
+We will only define a pre-sheaf of rings, however it's possible to define a
+pre-sheaf of sets, pre-sheaf of abelian groups, etc.
```

Comment:
somehow I had a misconception "sheaf must consist of rings", and this caused me a fair bit of confusion. For instance:

* when I see the mention of "quasicoherent sheaf" -- the important adjective is that they're sheaf of **modules**, not that they're quasicoherent (which is in some sense also important but to a lesser extent)
* exercise 2.3.J in Vakil has an example where the image presheaf is not a sheaf -- in that case they must be treated as sheaf of **sets** instead of sheaf of rings (actually, you can probably make it into sheaf of abelian group) -- because exp is not a ring homomorphism.